src/theory/quantifiers/theory_quantifiers_type_rules.h

```diff
@@ -25,26 +25,58 @@ namespace cvc5 {
 namespace theory {
 namespace quantifiers {
 
+/**
+ * Type rule used for FORALL and EXISTS. Ensures the first argument is a
+ * bound variable list, the second argument has Boolean Type, and the third
+ * argument (if it exists) is an instantiation pattern list. Returns the
+ * Boolean type.
+ *
+ * Furthermore ensures that certain annotations (e.g. for INST_POOL) are well
+ * formed. In particular, instantiation pool annotations specify how to
+ * instantiate this quantified formula. These must specify n sets, where n
+ * is the number of variables of this quantified formula.
+ */
 struct QuantifierTypeRule
 {
   static TypeNode computeType(NodeManager* nodeManager, TNode n, bool check);
 };
 
+/**
+ * Type rule for bound variable lists. Ensures its children are bound variables,
+ * and returns the bound variable list type.
+ */
 struct QuantifierBoundVarListTypeRule
 {
   static TypeNode computeType(NodeManager* nodeManager, TNode n, bool check);
 };
 
+/**
+ * Type rule for instantiation patterns. This checks for a common mistake
+ * of using terms instead of term lists in pattern annotations, and returns
+ * the instantiation pattern type.
+ */
 struct QuantifierInstPatternTypeRule
 {
   static TypeNode computeType(NodeManager* nodeManager, TNode n, bool check);
 };
 
+/**
+ * A quantifier annotation, which returns the inst pattern type.
+ *
+ * Furthermore ensures well-formedness of instantiation attributes with two
+ * children, which must have a keyword specified as a constant string as the
+ * first child.
```

Comment:
And what as the second child?

Reply:
The remaining children are arbitrary, ill comment.